query-engine/connectors/sql-query-connector/src/ordering.rs

```diff
@@ -17,172 +17,178 @@ pub struct OrderByDefinition {
-        _ => unreachable!("Order by relation aggregation other than count are not supported"),
-    };
-
-    // We perform the aggregation on the last join
-    let last_aggr_join = compute_aggr_join(
-        last_hop.into_relation_hop().unwrap(),
-        aggregation_type,
-        None,
-        ORDER_AGGREGATOR_ALIAS,
-        join_prefix.as_str(),
-        joins.last(),
-    );
-
-    // This is the final column identifier to be used for the scalar field to order by.
-    // `{last_join_alias}.{ORDER_AGGREGATOR_ALIAS}`
-    let order_by_column = Column::from((last_aggr_join.alias.to_owned(), ORDER_AGGREGATOR_ALIAS.to_owned()));
-
-    joins.push(last_aggr_join);
-
-    (joins, order_by_column)
-}
+    pub fn compute_joins_scalar(&mut self, order_by: &OrderByScalar) -> (Vec<AliasedJoin>, Column<'static>) {
+        let mut joins = vec![];
 
-pub fn compute_joins_scalar(
-    order_by: &OrderByScalar,
-    order_by_index: usize,
-    base_model: &ModelRef,
-) -> (Vec<AliasedJoin>, Column<'static>) {
-    let join_prefix = format!("{}{}", ORDER_JOIN_PREFIX, order_by_index);
-    let joins = order_by
-        .path
-        .iter()
-        .map(|hop| compute_one2m_join(base_model, hop.into_relation_hop().unwrap(), join_prefix.as_str()))
-        .collect_vec();
-
-    // This is the final column identifier to be used for the scalar field to order by.
-    // - If we order by a scalar field on the base model, we simply use the model's scalar field. eg:
-    //   `{modelTable}.{field}`
-    // - If we order by some relations, we use the alias used for the last join, e.g.
-    //   `{join_alias}.{field}`
-    let order_by_column = if let Some(last_join) = joins.last() {
-        Column::from((last_join.alias.to_owned(), order_by.field.db_name().to_owned()))
-    } else {
-        order_by.field.as_column()
-    };
-
-    (joins, order_by_column)
+        for (i, hop) in order_by.path.iter().enumerate() {
+            let previous_join = if i > 0 { joins.get(i - 1) } else { None };
+            let join = compute_one2m_join(hop.into_relation_hop().unwrap(), &self.join_prefix(), previous_join);
+
+            joins.push(join);
+        }
+
+        // This is the final column identifier to be used for the scalar field to order by.
+        // - If we order by a scalar field on the base model, we simply use the model's scalar field. eg:
+        //   `{modelTable}.{field}`
+        // - If we order by some relations, we use the alias used for the last join, e.g.
+        //   `{join_alias}.{field}`
+        let order_by_column = if let Some(last_join) = joins.last() {
+            Column::from((last_join.alias.to_owned(), order_by.field.db_name().to_owned()))
+        } else {
+            order_by.field.as_column()
+        };
+
+        (joins, order_by_column)
+    }
+
+    fn join_prefix(&mut self) -> String {
+        self.join_counter += 1;
+
+        format!("{}{}", ORDER_JOIN_PREFIX, self.join_counter)
+    }
```

Comment:
I introduced shared state in the order by renderer to make sure we always generate unique join aliases.